src/core/thread/mle_router.cpp

```diff
@@ -1492,6 +1495,18 @@ bool MleRouter::HasNeighborWithGoodLinkQuality(void) const
     return haveNeighbor;
 }
 
+void MleRouter::ResetRouterDisallowed(void)
+{
+    mDisallowRouterRoleTimeout = 120;
+    LogInfo("mDisallowRouterRoleTimeout set from Network Data change");
+}
+
+void MleRouter::ResetRouterDisallowedOnInit(void)
```

Comment:
These two are only different because I wanted more logging when I was testing. Could be collapsed into one function